toolkit/authentication/src/main/java/com/arcgismaps/toolkit/authentication/Authenticator.kt

```diff
@@ -185,3 +358,38 @@ private fun Context.findActivity(): Activity {
     }
     throw IllegalStateException("Could not find an activity from which to launch client certificate picker.")
 }
+
+/**
+ * Represents different types of browser-based authentication challenges.
+ *
+ * This sealed class is used to encapsulate specific authentication challenges that require
+ * interaction with a browser.
+ *
+ * @since 200.8.0
+ */
+public sealed class BrowserAuthChallenge {
```

Comment:
Better to spell this out:
```suggestion
public sealed class BrowserAuthenticationChallenge {
```